### PR TITLE
fix: support ip:port in host

### PR DESCRIPTION
**Pull request by theweakgod**
support ip:port in host
[#138](https://github.com/api7/lua-resty-radixtree/issues/138)
[#10531](https://github.com/apache/apisix/issues/10531)

### CODE REVIEW

File: lib/resty/radixtree.lua [hunk 1]
```diff
@@ -739,6 +739,11 @@ local function match_route_opts(route, opts, args)
         if host then
             local len = #hosts
             for i = 1, len, 2 do
+                if str_find(hosts[i+1], ":", 1, true) then
+                    if opts.vars.http_host then
+                        host = opts.vars.http_host
```
Comment (haoyang1994) [inline lib/resty/radixtree.lua:744]: 1. `host` should be reset to `opts.host` before next loop. 
2. `str_find(hosts[i+1], ":", 1, true)` could be done before route inserted into the radixtree like processing 'wildcard' for better performance.

File: lib/resty/radixtree.lua [hunk 2]
```diff
@@ -739,6 +739,13 @@ local function match_route_opts(route, opts, args)
         if host then
             local len = #hosts
             for i = 1, len, 2 do
+                if str_find(hosts[i+1], ":", 1, true) then
```
Comment (shreemaan-abhishek) [inline lib/resty/radixtree.lua:742]: what about hosts[i]?

File: lib/resty/radixtree.lua [hunk 3]
```diff
@@ -739,6 +739,13 @@ local function match_route_opts(route, opts, args)
...
+                    if opts.vars.http_host then
```
Comment (shreemaan-abhishek) [inline lib/resty/radixtree.lua:743]: why introduce a new variable?

**Review by Sn0rt — APPROVED** [review]

**Comment by monkeyDluffy6017** [comment]
I will check this later, thanks for your contribution!

**Comment by monkeyDluffy6017** [comment]
@theweakgod are you stilling working on the pr?

**Comment by theweakgod** [comment]
> @theweakgod are you stilling working on the pr?

Sure

**Comment by monkeyDluffy6017** [comment]
@theweakgod please resolve the review comments

**Comment by theweakgod** [comment]
> @theweakgod please resolve the review comments

what should I do, I already responsed

**Comment by theweakgod** [comment]
> @theweakgod please resolve the review comments

Ok, I got it

**Comment by monkeyDluffy6017** [comment]
@theweakgod I didn't see your reply
image

**Comment by theweakgod** [comment]
> @theweakgod I didn't see your reply image

image
But I really replied.

**Comment by monkeyDluffy6017** [comment]
The comment is still pending, please check the button at the upright corner
image

**Comment by theweakgod** [comment]
> The comment is still pending, please check the button at the upright corner image

Okay, thank you, this is my first time using this.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**